### ISSUE TITLE
Inference fails when running Faster RCNN model from ONNX model zoo

### ISSUE THREAD

**Issue by brantPTS** [microsoft/onnxruntime#1975]
**Describe the bug**
Running an inference with the Faster RCNN model from onnx model zoo fails

**Urgency**
We are blocked from using the ONNX runtime until this is fixed

**System information**
- OS Platform and Distribution: Windows 10 Pro
- ONNX Runtime installed from source (built locally with Sep 27, 2019 source code)
- Visual Studio version (if applicable): 2019
- CUDA/cuDNN version: 10.0 / 7.3
- GPU model and memory: nVidia GTX 1080ti

**To Reproduce**
Get latest source code (we used Sep 27 version)
Build locally with cuda similar to: 
.\build.bat --msvc_toolset 14.11 --use_cuda --cuda_home "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0" --cudnn_home C:\local\cudnn-7.3\cuda --build_csharp 

Download the Faster RCNN model from onnx model zoo (https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/faster-rcnn).
Ensure this file is on disk here: (or alter source code in unit test appropriately if file is in a different location): 
D:\modelsfrcnn\frcnn\faster_rcnn_R_50_FPN_1x.onnx 

Open the following solution in VS 2019: 
C:\local\onnx\onnxruntime\csharp\OnnxRuntime.CSharp.sln
Add a unit test to InferenceTest.cs (see code below) and run in debug mode:
C:\local\onnx\onnxruntime\csharp\test\Microsoft.ML.OnnxRuntime.Tests\InferenceTest.cs

The unit test code to add to InferenceTest.cs: 

```
[Fact]
private void frcnnTest()
{

	string modelsDir;
	// expect this file: D:\modelsfrcnn\frcnn\faster_rcnn_R_50_FPN_1x.onnx from https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/faster-rcnn
	modelsDir = @"D:\modelsfrcnn";
	var modelRoot = new DirectoryInfo(Path.Combine(modelsDir));
	foreach (var modelDir in modelRoot.EnumerateDirectories())
	{
		var onnxModelNames = modelDir.GetFiles("*.onnx");

		var onnxModelFileName = Path.Combine(modelsDir, modelDir.Name, onnxModelNames[0].Name);
		SessionOptions opts = new SessionOptions();
		//SessionOptions opts = SessionOptions.MakeSessionOptionWithCudaProvider(0);
		using (var session = new InferenceSession(onnxModelFileName, opts))
		{
			float[] dataIn;
			dataIn = new float[3 * 416 * 416];

			///// Create Inputs
			var nov = new List<NamedOnnxValue>();
			nov.Add(
				NamedOnnxValue.CreateFromTensor<float>
					("image", new DenseTensor<float>(dataIn, new int[] { 3, 416, 416 })));

			int count = 3;
			List<double> timesMs = new List<double>(count);
			Stopwatch sw = new Stopwatch();

			for (int i = 0; i < count; i++)
			{

				sw.Restart();
				///// Run Session
				using (var resnov = session.Run(nov))
				{
					timesMs.Add(sw.ElapsedMilliseconds);
					var res = resnov.ToArray()[0].AsTensor<float>().ToArray<float>();
				}
			}
			timesMs.Sort();
			double medianTime = timesMs[count / 2];
			Console.Out.WriteLine($"Median time: {medianTime:0.00} ms");
			Debug.WriteLine($"Median time: {medianTime:0.00} ms");

		}
	} //model
}
````

**Expected behavior**
The unit test should run without error or memory leaks. See screenshots below: 

![frcnnOnnxTestError](https://user-images.githubusercontent.com/34780180/66062061-891cd780-e4f5-11e9-9ff1-cf56d6aff1b1.JPG)

![frcnnOnnxTestErrorMemoryLeak](https://user-images.githubusercontent.com/34780180/66062071-8e7a2200-e4f5-11e9-94e1-601d87bec827.JPG)

**Comment by hariharans29** [comment]
Hi,

I think there are 2 dimensions to this issue:

1) I think when you feed it random data, the model is unable to make any detections
2) When the C# layer tries to create a tensor with no elements (number of detection boxes = 0), it throws. In Python, I get back an empty array of shape [0, 4].

So, please try following the demo in the link you pasted and try inferencing on an image where the model is sure to detect something. In legitimate cases when there is an image with no objects (or one that the model isn't able to detect), the C# layer must be able to handle it (addressed by #1976 )

**Comment by brantPTS** [comment]
@hariharans29  Thank you for working this so promptly.  I will work on preparing some real data and also will look forward to your fix for the zero dimension tensor.

**Comment by hariharans29** [comment]
Sure. The 0D tensor problem should be limited to just the c# layer (Python and C/C++ should be ok) - just FYI :)

**Comment by hariharans29** [comment]
Closing out this issue as the only takeaway for ORT was that the C# layer didn't handle creation of empty tensors (fixed via #1976) and for the provided input an empty tensor is the right output. So, the result is correct.
